wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -0,0 +1,142 @@
+    var isExpiringSoon: Bool  = false
+}
+
+enum E2EIdentityCertificateStatus: CaseIterable {
+    case notActivated, revoked, expired, valid, none
+}
+
+extension E2EIdentityCertificateStatus {
+    func titleForStatus() -> String {
+        switch self {
+        case .notActivated:
+            return "device.details.not.activated".localized
+        case .revoked:
+            return "device.details.revoked".localized
+        case .expired:
+            return "device.details.expired".localized
+        case .valid:
+            return "device.details.valid".localized
+        case .none:
+            return ""
+        }
+    }
+
+    func imageForStatus() -> Image {
+        switch self {
+        case .notActivated:
+            return Image(.certificateExpired)
+        case .revoked:
+            return Image(.certificateRevoked)
+        case .expired:
+            return Image(.certificateExpired)
+        case .valid:
+            return Image(.certificateValid)
+        case .none:
+            return Image(asset: .init(name: ""))
+        }
+    }
+}
+
+protocol DeviceDetailsViewActions {
+    func fetchCertificate() async
+    func showCertificate(validate: () -> Bool, result: (Bool) -> Void)
+    func removeDevice()
+    func resetSession()
+    func setVerified(_ result: (Bool) -> Void)
+    func copyToClipboard(_ value: String)
+}
+
+final class DeviceDetailsActionsHandler: DeviceDetailsViewActions {
+    var isDeviceVerified: Bool = false
+
+    func fetchCertificate() async {
+
+    }
+
+    func showCertificate(validate: () -> Bool, result: (Bool) -> Void) {
+        result(validate())
+    }
+
+    func removeDevice() {
+
+    }
+
+    func resetSession() {
+
+    }
+
+    func setVerified(_ result: (Bool) -> Void) {
+        result(isDeviceVerified)
+    }
+
+    func copyToClipboard(_ value: String) {
+        UIPasteboard.general.string = value
+    }
+
+}
+
+struct DeviceInfoViewModel {
+    let udid: String
+    let addedDate: String = Date().formattedDate
+    var title: String
+    let mlsThumbprint: String
+    let deviceKeyFingerprint: String
+    let proteusID: String
+    var isProteusVerificationEnabled: Bool
+    var e2eIdentityCertificate: E2EIdentityCertificate
+    let actionsHandler: DeviceDetailsViewActions
+
+    init(udid: String, title: String, mlsThumbprint: String, deviceKeyFingerprint: String, proteusID: String, isProteusVerificationEnabled: Bool, e2eIdentityCertificate: E2EIdentityCertificate, actionsHandler: DeviceDetailsViewActions = DeviceDetailsActionsHandler()) {
+        self.udid = udid
+        self.title = title
+        self.mlsThumbprint = mlsThumbprint
+        self.deviceKeyFingerprint = deviceKeyFingerprint
+        self.proteusID = proteusID
+        self.isProteusVerificationEnabled = isProteusVerificationEnabled
+        self.e2eIdentityCertificate = e2eIdentityCertificate
+        self.actionsHandler = actionsHandler
+        FontScheme.configure(with: .large)
+    }
+}
```

Comment:
Can we move that init into multiple lines? With Xcode 15 it's really easy to do using CTRL + M

Reply:
done